### PR TITLE
Add `Remote#PruneRef` method.

### PR DESCRIPTION
**Pull request by calavera**
And update libgit2 to support prunning.

### CODE REVIEW

**Comment by calavera** [comment]
This is blocked by either #152 or https://github.com/libgit2/libgit2/pull/2794 to be able to include `sys/openssl.h`.

File: remote.go [hunk 1]
```diff
@@ -595,6 +592,10 @@ func (o *Remote) UpdateFetchHead() bool {
 	return C.git_remote_update_fetchhead(o.ptr) > 0
 }
 
+func (o *Remote) PruneRefs() bool {
+	return C.git_remote_prune_refs(o.ptr) > 0
+}
```
Comment (carlosmn) [inline remote.go]: This is called `C.git_remote_prune()`, we should also call the Go method `Remote.Prune()`.

**Comment by carlosmn** [comment]
You should be able to `#include <git2/sys/opnessl.h>` just under `<git2.h>` and have it work.

**Comment by carlosmn** [comment]
Nevermind, I misread what the openssl include issue was.

**Comment by calavera** [comment]
I'm closing this to start fresh from master, since the openssl issue is fixed there and libgit2 has been also updated.

File: remote.go (hunk 1 above)
Comment (calavera) [inline remote.go]: There are two methods, aren't there? one that retrieves the setting (this call), and another one that performs the prune, the one that you mention. Should I call the setting getter something different?

Comment (carlosmn) [inline remote.go]: Oh right, we did leave the getter in; carry on.

**Pull request closed without merging**